### PR TITLE
:sparkles: Adding kai-analyzer-rpc server code

### PR DESCRIPTION
**Pull request by shawn-hurley**

### CODE REVIEW

File: kai-analyzer/provider/java/internal_java_provider.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package java
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/konveyor/analyzer-lsp/provider"
+	extjava "github.com/shawn-hurley/java-external-provider/pkg/java_external_provider"
```
Comment (djzager) [inline kai-analyzer/provider/java/internal_java_provider.go:8]: Is there something special in your provider?

```suggestion
	extjava "github.com/konveyor/analyzer-lsp/external-providers/java-external-provider"
```

File: kai-analyzer/provider/java/internal_java_provider.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	svcClient, _, err := p.Init(ctx, log, provider.InitConfig{
+		Location: location,
+		ProviderSpecificConfig: map[string]interface{}{
+			"lspServerName": "java",
+			"bundles":       "/Users/shurley/repos/MTA/java-analyzer-bundle/java-analyzer-bundle.core/target/java-analyzer-bundle.core-1.0.0-SNAPSHOT.jar",
+			"lspServerPath": "/Users/shurley/repos/kai/jdtls/bin/jdtls",
+		},
+		Proxy:        &provider.Proxy{},
+		AnalysisMode: "source-only",
+	})
```
Comment (djzager) [inline kai-analyzer/provider/java/internal_java_provider.go:30]: I would think bundles, lspServerPath, and analysisMode should be args to this.

Comment (djzager) [reply kai-analyzer/provider/java/internal_java_provider.go:30]: If it's easier to tackle this when moved into `initialize`, I am okay with that.

**Review by djzager — APPROVED** [review]
LGTM.

I added a couple of comments that I think could be addressed but both could be addressed in a follow-up when we create/handle the initialize, deactivate, exit methods.

File: kai-analyzer/provider/java/internal_java_provider.go (hunk 1 above)
Comment (shawn-hurley) [reply kai-analyzer/provider/java/internal_java_provider.go:8]: I fixed a bug, that emily fixed, need to update

File: kai-analyzer/provider/java/internal_java_provider.go (hunk 2 above)
Comment (shawn-hurley) [reply kai-analyzer/provider/java/internal_java_provider.go:30]: Yeah I think that needs to be in initialize, we need to add that. Do folks mind if that is a follow up?

**Review by JonahSussman — APPROVED** [review]
ACK

**Pull request merged**